packages/pool/contracts/RewardUtils.sol

```diff
@@ -22,10 +22,13 @@ abstract contract RewardUtils is GetterUtils, IRewardUtils {
             if (api3Token.getMinterStatus(address(this)))
             {
                 uint256 rewardAmount = totalStake * apr * EPOCH_LENGTH / 365 days / HUNDRED_PERCENT;
+                assert(block.number <= MAX_UINT32);
+                assert(rewardAmount <= MAX_UINT224);
                 epochIndexToReward[currentEpoch] = Reward({
                     atBlock: uint32(block.number),
                     amount: uint224(rewardAmount),
-                    totalSharesThen: totalShares()
+                    totalSharesThen: totalShares(),
+                    totalStakeThen: totalStake
```

Comment:
Another alternative could be publish a "RewardMinted" event and keep the totalStakeThen there for anyone to query.

Reply:
This information can already be derived from the events, this is for the data to be on-chain, usable by other contracts.
Say one wants to airdrop tokens to API3 stakers 1:1 for each API3 token they have staked at a point in time. We normally never record historical `totalStake` so this can only be done for the current block. This provides such use-cases with milestones they can use for that kind of stuff. Not really necessary, but doesn't hurt either.
I'll still add it as a field to `RewardMinted` though, that will make it easier to calculate `totalStake` off-chain at a point in time easier.